### ISSUE TITLE
Upgrade to GCC10 for better C++2a support

### ISSUE THREAD

**Issue by AeroStun** [DIT112-V20/group-09#67]
See `<compare>` and `<span>`

**Comment by platisd** [comment]
I was trying to use `std::span` with GCC 9.2 and it wouldn't be there. Good to know it came with GCC 10. I was waiting for it! :tada:

**Comment by AeroStun** [comment]
> https://en.cppreference.com/w/cpp/compiler_support

GNU libstdc++ with GCC10, LLVM libc++ with Clang/LLVM7, MS STL with MSVC19.26, AppleClang libc++ with XCode10

Technically we already use the pre-c++2a `gsl::span`, so that transition will be seamless for us

**Comment by platisd** [comment]
Were you using [Microsoft's implementation](https://github.com/Microsoft/GSL) for GSL?

**Comment by AeroStun** [comment]
Correct; however I mostly included the GSL for the its general purpose: a support library for the CppCoreGuidelines
If you're looking for a portable way to have spans in any ISO C++ version, I can recommend [span_lite](https://github.com/martinmoene/span-lite)

**Comment by platisd** [comment]
> Correct; however I mostly included the GSL for the its general purpose: a support library for the CppCoreGuidelines
> If you're looking for a portable way to have spans in any ISO C++ version, I can recommend [span_lite](https://github.com/martinmoene/span-lite)

That's exactly what I was after, thanks! :+1:
